src/lib/utils/memory_usage.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+
+namespace opossum {
+
+class MemoryUsage final {
```

Comment:
I can do STRONG_TYPEDEF(MemoryUsageInBytes, uint64_t); if that is what everybody wants.... :( If you're saying that I should just return uint64_t from estimate_memory_usage(): <insert random blog/talk about why you should use strong types here>

Reply:
> The type `size_t` is an implementation-defined unsigned integer type that is large enough to contain the size in bytes of any object.

The standard says that this is what you should use if you are expressing "the size in bytes of any object".

Reply:
(Sorry, I meant STRONG_TYPEDEF(MemoryUsageInBytes, size_t);)

So I should use size_t and introduce a method like format_bytes(size_t) that is essentially MemoryUsage::print()?

Reply:
> What is this good for? Do you expect MemoryUsage to be something different than bytes?

I asked myself the same.

Reply:
You two sure are a thoughtful bunch... I'd much prefer receiving clear statements in the form of "I'd prefer you do (precisely) ... if your intention here was ..." instead of vague chatter, because while I can guess, even 5 comments later I have no down to the point idea what you actually want me to replace "class MemoryUsage" and "print" with.


okay, gonna change this to size_t + free function 'format_bytes()'.